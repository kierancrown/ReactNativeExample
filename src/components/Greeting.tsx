import React from 'react';
import {StyleSheet, useColorScheme, View} from 'react-native';
import {Text} from 'react-native';
import moment from 'moment';

const Greeting = ({datetime = new Date()}) => {
  const isDarkMode = useColorScheme() === 'dark';

  const styles = StyleSheet.create({
    wrapper: {
      marginTop: 20,
    },
    text: {
      fontSize: 28,
      fontWeight: '700',
      color: isDarkMode ? '#fff' : '#000',
    },
    date: {
      fontSize: 14,
      color: '#a9a9a9',
    },
  });

  const timeAppropriateGreeting = (): string => {
    var today = datetime;
    var curHr = today.getHours();

    if (curHr < 12) {
      return 'Good morning';
    } else if (curHr < 18) {
      return 'Good afternoon';
    } else {
      return 'Good evening';
    }
  };

  const getReadableDate = (): string => {
    return moment(datetime).format('ddd, MMMM Do');
  };

  return (
    <View style={styles.wrapper}>
      <Text style={styles.date}>{getReadableDate()}</Text>
      <Text style={styles.text}>{timeAppropriateGreeting()},</Text>
      <Text style={styles.text}>Kieran</Text>
    </View>
  );
};

export default Greeting;
