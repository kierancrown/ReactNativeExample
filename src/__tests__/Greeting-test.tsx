import * as React from 'react';
import Greeting from '../components/Greeting';
import renderer from 'react-test-renderer';

const mockProps = {
  datetime: new Date('2020-01-01T00:00:00.000Z'),
};

it('renders correctly', () => {
  const tree = renderer.create(<Greeting {...mockProps} />);
  expect(tree).toMatchSnapshot();
});
