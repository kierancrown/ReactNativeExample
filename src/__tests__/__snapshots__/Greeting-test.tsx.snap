// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#a9a9a9",
        "fontSize": 14,
      }
    }
  >
    Wed, January 1st
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 28,
        "fontWeight": "700",
      }
    }
  >
    Good morning
    ,
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 28,
        "fontWeight": "700",
      }
    }
  >
    Kieran
  </Text>
</View>
`;
